Go 1.18 Survey

CamberLoid opened the pull request:
Topic Description
-----------------

This topic will update `go` to ~~1.18~~ 1.18.1. Check updates and rebuild all packages of `groups/go-rebuilds`.
<!-- Please input topic description here. -->

Package(s) Affected
-------------------

`go`
`groups/go-rebuilds`
`extra-base/golang-base`

And others
`scons` `flatbuffers`

cmdline: `ciel clean;ciel build -i {$instance} go go-md2man groups/go-rebuilds`
<!-- Please list all package(s) affected by this topic here. -->

Security Update?
----------------

No. 
<!-- If this topic contains security update(s), please uncomment "Yes,"
     mark with the `security` and `priority` labels, and make sure to mark your commits to relevant issue numbers.

     Please see GitHub's documentation on "Linking a pull request to an issue":

     https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue -->

<!-- Yes - Issue Number: ISSUENUMBER -->
<!-- No -->

<!-- Please uncomment the "Build Order" section if your topic affects more than one package. -->


Build Order
-----------

0. `scons` Noarch dependency. build and push first;
1. `go` The compiler itself;
2. `golang-base` `go-md2man` Toolkits for go;
3. `groups/go-rebuilds` All other packages;
4. `flatbuffers` Non-go-based package.

Test Build(s) Done
------------------

The checklist: https://github.com/AOSC-Dev/aosc-os-abbs/pull/3923#issuecomment-1120379096

**Primary Architectures**

<!-- Please remove any architecture to which this topic does not apply. -->

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`
<!-- If this package involves a `+32` counterpart, please uncomment the line below. -->    
<!-- - [ ] 32-bit Optional Environment `optenv32` -->

<!-- If all package(s) affected by this topic is `noarch`, please use the stub below. -->
<!-- - [ ] Architecture-independent `noarch` -->

**Secondary Architectures**

<!-- Please remove any architecture to which this topic does not apply. -->

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [x] Loongson 3 `loongson3`
- [x] ~~PowerPC 64-bit (Little Endian) `ppc64el`~~
- [x] RISC-V 64-bit `riscv64`

<!-- Maintainers should review file changes and, if the build script(s) affected complies with the
     [Package Styling Manual](https://wiki.aosc.io/developer/packaging/package-styling-manual/), please
     add `lgtm` label to this issue. -->

<!-- Maintainers and users may now test the packages in this topic and, once user/maintainer feedback indicates that the update(s) work as expected and find its quality satisfactory,
     another maintainer may now review this pull request and mark it as Approved. After which, the maintainer will build affected package(s) and upload them to the `stable` repository. -->

Update(s) Uploaded to Stable
----------------------------

**Primary Architectures**

<!-- Please remove any architecture to which this topic does not apply. -->

- [ ] AMD64 `amd64`   
- [ ] AArch64 `arm64`
<!-- If this package involves a `+32` counterpart, please uncomment the line below. -->
<!-- - [ ] 32-bit Optional Environment `optenv32` -->

<!-- If all package(s) affected by this topic is `noarch`, please use the stub below. -->
<!-- - [ ] Architecture-independent `noarch` -->

**Secondary Architectures**

<!-- Please remove any architecture to which this topic does not apply. -->

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] ~~PowerPC 64-bit (Little Endian) `ppc64el`~~
- [ ] RISC-V 64-bit `riscv64`

<!-- TODO: CI to auto-fill architectural progress. -->

CamberLoid committed:
go: update to 1.18
- go tools and net: branch switched to internal-branch.go1.18-vendor

CamberLoid committed:
delve: update to 1.8.2
... also rebuild for go 1.18

CamberLoid committed:
dep: rebuild for go 1.18 ...
... also bump REL 13 -> 14

CamberLoid committed:
glide: rebuild for go 1.18 ...
... also bump REL 13 -> 14

CamberLoid committed:
groups/go-rebuilds: add extra-devel/cf-tools

CamberLoid committed:
groups/go-rebuilds: reordered extra-devel/cf-tools

CamberLoid committed:
groups/go-rebuilds: fix typo

CamberLoid committed:
golang-base: ...
- add go's debugger tool `delve` to PKGDES
- rename `godep` to `dep`

CamberLoid committed:
delve: adopt arm64 support
P.S. upstream said it supports

CamberLoid committed:
gotop: update to 4.1.3

CamberLoid committed:
go: (ppc64el) turn on data race detector
... Requires further testing

CamberLoid committed:
go: bump REL 0 -> 1 in response of ...
... changes for ppc64el

CamberLoid committed:
golang-base: move `delve` to amd64's pkgdep

CamberLoid commented:
Upstream has released newer go compiler 1.18.1 . Previously updated packages will be rebuilt.

CamberLoid committed:
go: update to 1.18.1

CamberLoid committed:
go: make sure target is right when building for amd64

CamberLoid committed:
dep: rebuild for go 1.18.1

CamberLoid committed:
glide: rebuild for go 1.18.1

CamberLoid committed:
gotop: rebuild for go 1.18.1

CamberLoid committed:
delve: update to 1.8.3

CamberLoid committed:
popub: update to 0+git20210814
- dropped gccgo compiler patch, as gccgo may cause some issues;
- enabled position-independent code for all arches except loongson3

CamberLoid committed:
go: update package dependencies
- Building packages when enabling Positiona-independent executables requires llvm

CamberLoid committed:
go: revert previous modification for riscv64

CamberLoid committed:
popub: update anitya id

CamberLoid committed:
frp: update to 0.42.0

CamberLoid commented:
`popub` is tested and perform well on `riscv64`.
Despite there is no major behavior change between two commit of upstream, this package may require further tests due to the change of compiler from gccgo to google-go

CamberLoid commented:
Test checklist for `frp` | in $SRCDIR `(uname -a; HOME=/root make test) | curl -F "c=@-" "http://fars.ee/"`
- [x] amd64 | Log: http://fars.ee/CjeV
- [x] arm64 | Log: http://fars.ee/chDl
- [x] loongson3 | Log: http://fars.ee/eYJ4
- [x] ppc64el | http://fars.ee/vecw
- [x] riscv64 | Log: http://fars.ee/HvRr

CamberLoid committed:
clash: update to 1.10.6

CamberLoid commented:
Test checklist for hugo | Can successfully host https://github.com/AOSC-Dev/aosc-portal-kiss.github.io , and try to access it via `curl -L` or `lynx` (modern browser if possible)
- [x] amd64 
![image](https://user-images.githubusercontent.com/22998116/167298422-38d0472f-dc6b-48dc-9885-1989ca62ed02.png)
- [x] arm64
![image](https://user-images.githubusercontent.com/22998116/167300201-ea20f22d-8351-4531-97b3-3a6d249f4bf7.png)
- [x] ppc64el
![image](https://user-images.githubusercontent.com/22998116/167300777-5153e55e-405c-4d1b-acfc-edf2a707436b.png)
- [x] loongson3
![image](https://user-images.githubusercontent.com/22998116/167300686-c21548c3-007a-43f9-a6da-a2e37f83fcbb.png)
- [x] riscv64 
![image](https://user-images.githubusercontent.com/22998116/167299144-f99a1274-5055-4b7b-b55c-cd7c5ed830f4.png)

CamberLoid committed:
git-lfs: update to 3.1.4

CamberLoid committed:
gh/github-cli: Update to 2.9.0 with multiple changes
- Merge `github-cli` to `gh`, as these two packages are the same
  - `build` from `github-cli` is merged into `beyond` from `gh`
  - Change `PKGSEC` of `github-cli` to `vcs`
  - `github-cli` is now a dummy package with dependency of `gh`

CamberLoid committed:
go-md2man: Update to 2.0.2

CamberLoid commented:
Test checklist for packages, so far
1. Core component
- [x] The compiler itself `go` | As long as there is no building error for other packages, considering this is tested
2. Packages to upgrade
- [x] `delve` 
- [x] `frp` https://github.com/AOSC-Dev/aosc-os-abbs/pull/3923#issuecomment-1120217271
- [x] `hugo` https://github.com/AOSC-Dev/aosc-os-abbs/pull/3923#issuecomment-1120233524
- [x] `gotop` https://github.com/AOSC-Dev/aosc-os-abbs/pull/3923#issuecomment-1120590282 | Functionality tests are required as this upgrade enables `-buildmode=pie`
- [ ] ~~`clash`~~ https://github.com/AOSC-Dev/aosc-os-abbs/pull/3923#issuecomment-1121127189 | Skipped/Delayed
- [x] `popub` | Proved functionality on riscv64. 
- [ ] `git-lfs` https://github.com/AOSC-Dev/aosc-os-abbs/pull/3923#issuecomment-1148260409
- [x] `gh`/`github-cli` https://github.com/AOSC-Dev/aosc-os-abbs/pull/3923#issuecomment-1121950702
- [x] `go-md2man` | Check if `gh` is successfully built.
- [ ] `rclone` | Skipped as its unit test requires compile without LTO.
- [ ] `xray` https://github.com/AOSC-Dev/aosc-os-abbs/pull/3923#issuecomment-1148588022 | Functionality tests are required as this upgrade enables `-buildmode=pie`
- [ ] `v2ray` https://github.com/AOSC-Dev/aosc-os-abbs/pull/3923#issuecomment-1148577048 | Functionality tests are required as this upgrade enables `-buildmode=pie`
- [ ] `keybase` | May open a new topic to package other co
- [ ] `geoipupdate` | Functionality tests are required as this upgrade enables `-buildmode=pie`
- [ ] `ccat` https://github.com/AOSC-Dev/aosc-os-abbs/pull/3923#issuecomment-1148603458
- [x] `hub` | Functionality tests are required as this upgrade enables `-buildmode=pie` | https://github.com/AOSC-Dev/aosc-os-abbs/pull/3923#issuecomment-1121837159
- [ ] `kcptun` | Upstream does not provide unit test | Functionality tests are required as this upgrade enables `-buildmode=pie`
- [ ] ~~`flatbuffers`~~ | Skipped. Will be dropped out of go-rebuilds.
- [ ] `syncthing` https://github.com/AOSC-Dev/aosc-os-abbs/pull/3923#issuecomment-1148621989
- [ ] `baidupcs-go` | Minor update. May not be tested.
- [x] `pcstat` | Switch buildtype to ABTYPE.
- [ ] `gogs` | https://github.com/AOSC-Dev/aosc-os-abbs/pull/3923#issuecomment-1128624326 | Functionality tests are required as this upgrade enables `-buildmode=pie` 
- [ ] `direnv` | https://github.com/AOSC-Dev/aosc-os-abbs/pull/3923#issuecomment-1148238124 | Functionality tests are required as this upgrade enables `-buildmode=pie`
- [ ] `gost` | Upstream not provide / Tests are broken | Functionality tests are required as this upgrade enables `-buildmode=pie`
- [ ] `android-platform-tools` | Minor update; Tests are not required.
- [ ] `elvish` | https://github.com/AOSC-Dev/aosc-os-abbs/pull/3923#issuecomment-1148171959 | Major changes are applied. Requires tests.
- [ ] `pup` https://github.com/AOSC-Dev/aosc-os-abbs/pull/3923#issuecomment-1148648223
- [ ] `fzf` https://github.com/AOSC-Dev/aosc-os-abbs/pull/3923#issuecomment-1148639835
- [ ] `restic` https://github.com/AOSC-Dev/aosc-os-abbs/pull/3923#issuecomment-1148662131
- [ ] `mongodb` | Build failed due to changes to glibc 2.34 |Database upgrade may require additional user configuration 
- [ ] `scons` | Builddep of MongoDB 4.4.14 . Not a go package.
3. Packages to rebuild
- [ ] `libguestfs`: Too complex. Will not upgrade in this topic (or after all other main upgrade complete)
- [ ] `v2ray-plugins` | Functionality tests are required because this update switches build method to ABTYPE
- [ ] `obfs4proxy`
- [ ] `dep`
- [ ] `glide`
- [ ] `podman` | Upgrade of which may be proceeded in next OCI/container survey;

CamberLoid committed:
groups/go-rebuilds: Remove github-cli

CamberLoid committed:
libguestfs: Bump REL; Rebuild for go 1.18/1.18.1

CamberLoid committed:
mongodb: Bump REL; Rebuild for go 1.18/1.18.1

CamberLoid committed:
xray: Update to 1.5.5
This update enables `buildmode=pie`. Further functionality tests are required.

CamberLoid committed:
v2ray: Update to 4.45.0
This update enables `-buildmode=pie`. Further functionality tests are required.

CamberLoid committed:
v2ray-plugin: Bump REL; Rebuild for go 1.18/1.18.1
This commit switches its build methon to ABTYPE, which enables `-buildmode=pie` ...
... Further tests are required.

CamberLoid committed:
keybase: Update to 5.9.3
- Switch SRCS type to tarball due to excessive size of git repository

CamberLoid committed:
geoipupdate: update to 4.9.0
This update enables `-buildmode=pie`. Further functionality tests are required.

CamberLoid committed:
obfs4proxy: Bump REL; Rebuild for go 1.18/1.18.1

CamberLoid committed:
ccat: update to 1.1.0+git20201007

CamberLoid committed:
hub: update to 2.14.2
This update enables `-buildmode=pie` for supported arch and ...
... enables `-trimpath` by setting GOFLAGS
Further tests are required

CamberLoid committed:
kcptun: update to 20210922
This update enables `-buildmode=pie` for supported arch and ...
... enables `-trimpath` by setting GOFLAGS
Further functionality tests are required

CamberLoid commented:
Test checklist for `gotop` | Pass if normally executed.
(Reset as new patch introduced)
- [x] amd64
![image](https://user-images.githubusercontent.com/22998116/167361517-a372e30d-29f2-44d1-be2d-89fdcc45a464.png)
- [x] arm64
![image](https://user-images.githubusercontent.com/22998116/167361883-220ee360-2cec-455f-998b-b5d8bdd6cb53.png)
- [x] loongson3
![image](https://user-images.githubusercontent.com/22998116/167366511-dd54848a-8699-4346-ac85-ee2a2f8a8b59.png)
- [x] ppc64el
![image](https://user-images.githubusercontent.com/22998116/167366397-e8367bf6-a994-44c1-ac64-7f55c1dc1b4e.png)
- [x] riscv64
![image](https://user-images.githubusercontent.com/22998116/167355500-8daf6c10-4186-4469-9edc-ec23b9e161e6.png)

CamberLoid committed:
gotop: Fix an issue that gotop version cannot be ...
... correctly displayed by `gotop -v`; Bump REL; Enable `-buildmode=pie` ...
... to harden binary for supported architecture and `-trimpath` for ...
... reproducible build;

CamberLoid committed:
gotop: fix FTBFS on riscv64

CamberLoid committed:
gotop: Fix build tag syntax error

CamberLoid committed:
gotop: Enable generating of manpage; Bump REL

CamberLoid committed:
gotop: Fix an unexpected typo

CamberLoid committed:
gotop: Fix another unexpected typo

CamberLoid commented:
Patches in commit [0a71c63](https://github.com/AOSC-Dev/aosc-os-abbs/pull/3923/commits/0a71c63df0b405818070e4615b0b306270c64108) and [ca7dfe9](https://github.com/AOSC-Dev/aosc-os-abbs/pull/3923/commits/ca7dfe94787a551e26e3c454e530ea9da0dc543b) have been formed into a pull request and submitted to upstream. This patch will solve a FTBFS issue on riscv64 platform.

Reference: 
- https://github.com/xxxserxxx/gotop/pull/218
- https://github.com/xxxserxxx/gotop/pull/218/commits/658926bd0fa8e15c75c4db1426f2b20307312870

CamberLoid committed:
clash: Enable `-buildmode=pie`

CamberLoid commented:
Tests for clash(FOSS) are skipped because a checking dependency `docker` is missing, which is required by upstream-shipped test suite.

CamberLoid committed:
hub: Update to 2.14.2+git20220404
- Switch SRCS type to git as many bugs are fixed since last release
which is 2 years ago

CamberLoid commented:
Test for `hub` will be performed by adding these lines to `autobuild/beyond` (will be discarded in formal build). Check if no error reports.
```bash
abinfo "Checking ..."
pushd "$SRCDIR"/src/github.com/github/hub
    make test || abwarn "Test failed ..."
popd
```
- [x] amd64 | Log: https://paste.aosc.io/paste/2dY4lujv5PMWcKV7uOWDbg 
![image](https://user-images.githubusercontent.com/22998116/167547984-0c378610-c30d-4ebd-b095-02796f4cc6c7.png)
- [x] arm64
![image](https://user-images.githubusercontent.com/22998116/167548286-d22363fe-4d84-4332-a861-2936c5889de7.png)
- [x] loongson3
![image](https://user-images.githubusercontent.com/22998116/167553858-978c4849-f1f5-42c0-9c9a-d6a41a9d86bf.png)
- [x] ppc64el
![image](https://user-images.githubusercontent.com/22998116/167554263-6d29b616-b329-44ad-be7a-eb66afef7d6a.png)
- [x] riscv64
![image](https://user-images.githubusercontent.com/22998116/167554832-6109273b-7453-43b2-8c27-09282f0b29e9.png)

CamberLoid committed:
hub: Add a postinst as a reminder to user

CamberLoid committed:
hub: Remove unneeded line in `build`

CamberLoid commented:
Tests for `gh` will be performed using integrated check by adding testing to `autobuild/beyond` (will be restored). Check if build succeeds with no error.
cmdline:
```bash
(echo -e "\
abinfo \"Checking ...\"
(make test || abdie \"Test failed.\") && abinfo \"Test passed.\"
" >> extra-vcs/gh/autobuild/beyond ) && ciel clean && ciel build -i go-survey gh && git restore . && ciel clean && ciel build -i go-survey gh
```
- [x] amd64 (pushed)
- [x] arm64
- [x] loongson3
- [x] ppc64el
- [x] riscv64

CamberLoid committed:
gh: Set GOFLAGS including buildmode=pie, trimpath
- Also set linkmode to external to avoid test-phase error (make test)

CamberLoid committed:
gh: fix a typo

CamberLoid committed:
github-cli: mark ABHOST=noarch

CamberLoid committed:
geoipupdate: Replace pandoc with architecture-independent ronn-ng and re-enable
manpage generation for haskell-unavailable architecture.
- Update dependencies list

CamberLoid committed:
flatbuffers: Update to 2.0.6;
- Now packaging and shipping with python2/3 library;
- Enable building for shared library

CamberLoid committed:
flatbuffers: Drop builddep `openjdk` for riscv64

CamberLoid committed:
flatbuffers: Refine builddep list

CamberLoid commented:
Unit tests for `flatbuffers` are skipped as built-in tests are broken (also reported by archlinux).

CamberLoid committed:
syncthing: Update to 1.20.1

CamberLoid committed:
pcstat: Update to 0.0.1; Switch building method to ABTYPE

CamberLoid committed:
pup: Rebuild for go 1.18/1.18.1; Bump REL

CamberLoid committed:
flatbuffers: Drop builddep of go and openjdk

CamberLoid commented:
`flatbuffers` does not require `go` and `openjdk` to build. Will be dropped out of `groups/go-rebuilds` by the end of survey.

CamberLoid committed:
podman: Rebuild for go 1.18/1.18.1; Bump REL
Upgrade of this package is postponed

CamberLoid committed:
gogs: Update to 0.12.6; Enable buildmode=pie
Along with some expression tweaks

CamberLoid committed:
android-platform-tools: Update to 31.0.3p1

CamberLoid committed:
direnv: Update to 2.30.3

CamberLoid committed:
gost: Update to 2.11.2

CamberLoid committed:
gogs: Fix missing export statement

CamberLoid committed:
direnv: Set GOFLAGS with buildmode=pie and others

CamberLoid committed:
gost: Set GOFLAGS with buildmode=pie and others

CamberLoid committed:
elvish: Update to 0.18.0; Major changes are applied:
- From version 0.16.0, elvish changes its file hierarchy, resulting a
FTBFS when ABTYPE is set to gomod. This commit contains a build script
to fix aforementioned problem.
- Due to the change of packaging process, unnecessary executables
are removed. Now only elvish itself is in /usr/bin
- Also include the source file (mostly markdown) of documentation.
As pandoc is not available on platform other than amd64, generation of
documentation is skipped.

CamberLoid committed:
elvish: Bugfixes; Bump REL
- Rename preinst to postinst (Well, it is a mistake)
- Fix installation path of documentations

CamberLoid committed:
groups/go-rebuilds: Drop flatbuffers

CamberLoid committed:
dnscrypt: Update to 2.1.1

CamberLoid committed:
graftcp: Update to 0.4.0

MingcongBai commented on extra-shells/elvish/autobuild/postinst:
```diff
@@ -0,0 +1,5 @@
+grep -Fqx '/bin/elvish' /etc/shells || \
+grep -Fqx '/usr/bin/elvish' /etc/shells || \
```
Try and add some indents after line breaks to denote a chain of commands.

```
grep -Fqx '/bin/elvish' /etc/shells || \
    grep -Fqx '/usr/bin/elvish' /etc/shells || \
    echo '/usr/bin/elvish' >> /etc/shells
```

No need to tell user about /etc/shells update.

MingcongBai commented on extra-shells/elvish/autobuild/postrm:
```diff
@@ -0,0 +1,8 @@
+# This command is adopted from Arch Linux's elvish.install
+# Reference: 
+# https://github.com/archlinux/svntogit-community/blob/packages/elvish/trunk/elvish.install
```
Use a specific commit snapshot for your reference.

MingcongBai commented on extra-shells/elvish/autobuild/postrm:
```diff
@@ -0,0 +1,8 @@
+# Reference: 
+# https://github.com/archlinux/svntogit-community/blob/packages/elvish/trunk/elvish.install
+if [ $1 = 'remove' ]; then
+    echo "Attempting to purge elvish from /etc/shells ..."
```
Again, no need to add outputs like this. We might address this in a shell survey for consistency.

MingcongBai commented on extra-shells/elvish/autobuild/prepare:
```diff
@@ -0,0 +1,6 @@
+abinfo "Setting GOFLAGS ..."
```
Be a bit more specific about what GOFLAGS you've set here, and move the all-arch export line directly after this line.

MingcongBai commented on extra-shells/elvish/autobuild/build:
```diff
@@ -0,0 +1,26 @@
+export GOPATH="$SRCDIR/abgopath"
+
+# The following command is partitially adopted from Arch Linux's PKGBUILD
```
"Adapt" *, and also by adapting a script, whether it's partial or not is less relevant. Minor note, but do fix your word choice.

MingcongBai commented on extra-shells/elvish/autobuild/build:
```diff
@@ -0,0 +1,26 @@
+export GOPATH="$SRCDIR/abgopath"
```
Missing abinfo.

MingcongBai commented on extra-shells/elvish/autobuild/build:
```diff
@@ -0,0 +1,26 @@
+    "$SRCDIR"/{README,SECURITY}.md
+
+abinfo "Installing documentations ..."
+mkdir -vp "$PKGDIR"/usr/share/doc/elvish/website
```
Why "website"? You could consider renaming the README.md above to ELVISH.md.

CamberLoid committed:
fzf: Update to 0.30.0
- Switch building method to ABTYPE=gomod
- Refine previous build script

CamberLoid committed:
elvish: Lint and refine build scripts; Bump REL

CamberLoid committed:
dnscrypt: Refine abinfo's output

CamberLoid committed:
gost: Refine abinfo's output and lint build script

CamberLoid committed:
frp: Set GOFLAGS for reproducible build and others

CamberLoid committed:
xray: Refine build script and abinfo's output

CamberLoid committed:
geoipupdate: Lint & refine build script;
Also set additional GOFLAGS

CamberLoid committed:
gh: Lint & refine build script and abinfo's output;
Also set additional GOFLAGS

CamberLoid committed:
v2ray-plugin: Set GOFLAGS for reproducible build and others

CamberLoid committed:
direnv: Refine abinfo's output and lint build script

CamberLoid committed:
kcptun: Lint & refine build script and abinfo's output;
Also set additional GOFLAGS

CamberLoid committed:
gotop: Lint & refine build script and abinfo's output;
Also set additional GOFLAGS

CamberLoid committed:
hub: Lint & refine build script and abinfo's output;
Also set additional GOFLAGS

CamberLoid committed:
restic: update to 0.13.1
- Refine build script by dropping outdated configuration;
- Lint build script in accordance with the Styling Manual

CamberLoid committed:
cf-tool: rebuild for go 1.18/1.18.1; bump REL

CamberLoid committed:
groups/go-rebuilds: update list

CamberLoid committed:
cf-tool: mark FAIL_ARCH="riscv64" due to the ...
... different availablity of syscall on riscv64 and other architecture
Also update groups/go-rebuilds.riscv64 .

CamberLoid committed:
pup: update to 0.4.0+git20220307
- Switch SRCS type to git and build method to ABTYPE=gomod,
as upstream has made tweaks reflecting changes to newer go compiler.

CamberLoid committed:
elvish: refine file hierarchy

CamberLoid committed:
scons: update to 4.3.0
Building of MongoDB uses new features of scons. Upgrading of
which will solve a FTBFS issue of scons.

CamberLoid committed:
mongodb: update to 4.4.14, with fixes of some vulnerabilities
- Refine build script
  - Refine abinfo's output
  - Transfer manipulations of mongo-tools from prepare to spec files

CamberLoid committed:
libguestfs: temporarily disable build for riscv64 ...
due to dependency problem.

CamberLoid committed:
mongodb: multiple fixes:
- Rename patches' filename in accordance to Styling manual;
- Introduce a new patch to fix a FTBFS due to missing library
- Fix wrong variable name in build script

CamberLoid committed:
dnscrypt: multiple changes:
- Update outdated patch
- Rename patch in accordance to the Styling Manual
- Update configurations in build script and conffile following
in accordance to upstream's change

CamberLoid committed:
dnscrypt: fix missing slash in build script

CamberLoid committed:
dnscrypt: conffile: strip empty lines

CamberLoid committed:
graftcp: fix FTBFS by backporting
- Also update systemd unit file to follow upstream's configuration

CamberLoid committed:
keybase: refine build script to solve a FTBFS problem;
Strip unneeded dependency

CamberLoid committed:
gost: mark FAIL_ARCH="riscv64" due depending ...
go module error.

CamberLoid committed:
gogs: update patches to latest upstream version(AUR)

CamberLoid committed:
rclone: remove deprecated funtion call

CamberLoid commented:
Test build for arm64 is successfully performed

CamberLoid committed:
kcptun: fix a FTBFS caused by outdated dependency

CamberLoid committed:
kcptun: minor expression refine and enable update of dependencies

CamberLoid committed:
v2ray-plugin: fix a FTBFS caused by outdated dependency

CamberLoid committed:
gogs: minor refinement of dependency list

CamberLoid commented:
Tests for `gogs` is successfully performed on the following platforms: `gogs web` + `lynx localhost:3000`
- [x] `amd64`
![image](https://user-images.githubusercontent.com/22998116/168776048-75377410-d313-4e84-9f25-af7054db9914.png)
- [x] `arm64`
![image](https://user-images.githubusercontent.com/22998116/172544921-e21aad45-0764-49fe-9b34-2e31de212c16.png)
- [ ] `loongson3`
- [ ] `riscv64`

OriginCode commented:
![image](https://user-images.githubusercontent.com/17294071/171789107-219c8951-ff97-47c6-ad9f-b35edabdd66d.png)

CamberLoid committed:
hugo: fix file belonging issue.

CamberLoid commented:
From this point, test will be performed via a patched version of autobuild3.

See also: 
- https://github.com/AOSC-Dev/autobuild3/pull/129

CamberLoid committed:
elvish: add FIXME reminder to rollback REL

CamberLoid commented:
Test checklist for `elvish`: 
- [x] `amd64`
![image](https://user-images.githubusercontent.com/22998116/172293607-72bbf956-e56f-4bf6-b658-dc785bbe2d02.png)
- [x] `arm64`
![image](https://user-images.githubusercontent.com/22998116/172544353-de6ee865-f90e-4e08-9c2a-c113944cd7b7.png)
- [ ] `loongson3`
- [ ] `riscv64`

Script: https://github.com/AOSC-Dev/aosc-os-abbs/commit/c69c4222f3ab7d0cec23c4cc59cae7ec0a40cdfb

CamberLoid committed:
direnv: drop pre-build GOFLAGS setting

CamberLoid commented:
Test checklist for `direnv`: 
- [x] `amd64` | Test exited with acceptable known failure
![image](https://user-images.githubusercontent.com/22998116/172308497-f61c51a5-1f8f-47c4-b5f9-a1b7dcfcdebb.png)
- [x] `arm64`
![image](https://user-images.githubusercontent.com/22998116/172544620-5a7fc3ad-4d46-472a-a3ed-ee3725aa9b2a.png)
- [ ] `loongson3`
- [ ] `riscv64`

Script: https://github.com/AOSC-Dev/aosc-os-abbs/commit/35e457be4a8f25db5ee7f3182d922b611a13e49e

CamberLoid commented:
Test checklist for `git-lfs`:
- [x] `amd64` no major failure
![image](https://user-images.githubusercontent.com/22998116/172313393-bf25211d-536c-4c4f-abdf-c2a430b1c19f.png)
- [x] `arm64` 
![image](https://user-images.githubusercontent.com/22998116/172544100-e12dc2cc-598f-4599-ac92-c9fb18a9396b.png)
- [ ] `loongson3`
- [ ] `riscv64`

CamberLoid committed:
gh: strip pre-build setting of GOFLAGS

Signed-off-by: Camber Huang <camber@poi.science>

CamberLoid committed:
rclone: strip redundant GOFLAGS

Signed-off-by: Camber Huang <camber@poi.science>

CamberLoid commented:
Unit test for `rclone` is skipped as it requires rclone to be built without LTO, which violates out package building principles.

CamberLoid commented:
pcstat: LGTM on amd64
![image](https://user-images.githubusercontent.com/22998116/172370594-135fdb1a-b2dd-49c0-8180-4e7940edf1fc.png)

CamberLoid commented:
v2ray: test checklist
- [x] `amd64` with a minor failure on TestTCPLocalNameServer
![image](https://user-images.githubusercontent.com/22998116/172375039-b5f31872-1d6a-49da-aab2-94a0963b2c11.png)
- [x] `arm64`
![image](https://user-images.githubusercontent.com/22998116/172788574-ca35c6dc-c343-47dd-8480-c766f5531d93.png)
- [ ] `loongson3`
- [ ] `riscv64`

CamberLoid commented:
xray:
- [x] `amd64` minor failure as geoip is not available, consider passed\
![image](https://user-images.githubusercontent.com/22998116/172376817-d155d81a-2738-42d0-986d-bcaf80fc2e45.png)
- [x] `arm64`
- [ ] `loongson3`
- [ ] `riscv64`

CamberLoid commented:
ccat:
- [x] `amd64`: seems good to me, at least it works
![image](https://user-images.githubusercontent.com/22998116/172379390-4f235758-8ce3-443b-81f6-6d3f8e901b5f.png)
- [ ] `arm64`
- [ ] `loongson3`
- [ ] `riscv64`

CamberLoid commented:
syncthing: 
- [x] `amd64`
![image](https://user-images.githubusercontent.com/22998116/172382587-b7a34cac-49db-4e14-860d-1fd118ad3933.png)
- [x] `arm64`
![image](https://user-images.githubusercontent.com/22998116/172801633-112a8975-4cfc-4f3e-9100-109eb84b0841.png)
- [ ] `loongson3`
- [ ] `riscv64`

CamberLoid commented:
fzf:
- [x] `amd64`
![image](https://user-images.githubusercontent.com/22998116/172385300-27e221ea-a247-44a3-863d-cb56079613f9.png)
- [x] `arm64`
- [ ] `loongson3`
- [ ] `riscv64`

CamberLoid commented:
pup: with an expected difference since https://github.com/ericchiang/pup/commit/6d1318bfa484b60309fe7a13e036093c179f15d6
- [x] `amd64`
![image](https://user-images.githubusercontent.com/22998116/172387163-2eb37fe5-9edc-4c55-b7c7-ed3cacb9f341.png)
- [x] `arm64`
![image](https://user-images.githubusercontent.com/22998116/172558596-f7cbb7e4-565a-4d40-9c4a-4cc85733a2e5.png)
- [ ] `loongson3`
- [ ] `riscv64`

CamberLoid commented:
restic: unit test failed as fusermount failed inside container. May need further test on non-container environment.
![image](https://user-images.githubusercontent.com/22998116/172389750-9789588d-7d9e-4864-a030-8edbda5b0247.png)
- [ ] `amd64`
- [ ] `arm64`
- [ ] `loongson3`
- [ ] `riscv64`

CamberLoid committed:
hugo: rebuild for go 1.18.1; bump rel; resolve confilcts.

CamberLoid commented:
Test for restic will be skipped and placed to next survey.
Marking this PR as ready for review

CamberLoid committed:
gotop: revert REL bump

Signed-off-by: Camber Huang <camber@poi.science>

CamberLoid committed:
dep: revert REL bump

Signed-off-by: Camber Huang <camber@poi.science>

CamberLoid committed:
glide: revert REL bump

Signed-off-by: Camber Huang <camber@poi.science>

CamberLoid committed:
elvish: revert REL bumping

Signed-off-by: Camber Huang <camber@poi.science>

CamberLoid committed:
golang-base: fix dependency on non ARM64/AMD64 platform

Signed-off-by: Camber Huang <camber@poi.science>
